### PR TITLE
iOS: remove Ti.Network.TCPSocket

### PR DESCRIPTION
**Pull request by sgtcoolguy**
**JIRA:** 
- https://jira.appcelerator.org/browse/TIMOB-27619
- https://jira.appcelerator.org/browse/TIMOB-27076

**Optional Description:**
We have kept around `Ti.Network.TCPSocket` since SDK 1.7 because it's used by the bonjour service code on iOS. (The new class is `Ti.Network.Socket.TCP`)

This PR attempts to finally remove it.

In addition, when moving to running on the main thread, our APIs were broken in that they'd block forever (`Ti.Network.BonjourService.publish()` and `Ti.Network.BonjourService.resolve()`). This PR removes the fiddling with run loops and usage of locks, rendering the `stop()`, `publish()` and `resolve()` methods async. In order to be notified of the results/completion of those methods, new `publish`, `stop` and `resolve` events were added. I also accept optional callback arguments to these functions. These operate like typical Node-style callback functions, with the first argument an optional `Error` (for error conditions), and the second argument being a boolean (for success conditions).

I've opted to retain both styles of usage since most of our APIs are event based, but longer-term we'd like to conform to Node.JS API styles more consistently.

Test code:
```javascript
var win = Ti.UI.createWindow({
    backgroundColor: '#fff'
});

var btn = Ti.UI.createButton({
    title: 'Trigger'
});

btn.addEventListener('click', function() {
    Ti.API.info('Hello world!');
    // Create the Bonjour Browser (looking for http)
    // See more Bonjour Types here: https://developer.apple.com/library/mac/qa/qa1312/_index.html
    const httpBonjourBrowser = Ti.Network.createBonjourBrowser({
        serviceType : '_http._tcp',
        domain : 'local'
    });

    // Handle updated services
    httpBonjourBrowser.addEventListener('updatedservices', function (e) {
        for (const service of e.services) {
            console.log(service.name);
            console.log(service.domain);
            console.log(service.type);
            console.log(service.isLocal);
            console.log(service.socket);
            // event style
            service.addEventListener('resolve', function (e) {
                console.log(e);
                console.log(service.socket);
                console.log(service.socket.port);
                console.log(service.socket.host);
            });
            // callback style
            service.resolve(120, (err, success) => {
                console.log(service.socket);
                console.log(service.socket.port);
                console.log(service.socket.host);
            });
        }
    });

    // Start searching
    httpBonjourBrowser.search();
});

win.add(btn);
win.open();
```

I *have* now tried with code I found at https://github.com/rborn/Advanced-Titanium-Tutorial---Bonjour-Networking/blob/master/Resources/bj.js as well, which creates a service and publishes it. I included an example in my commits for the docs for both browsing and resolving; and creating and publishing a service. Both worked for me locally.

### CODE REVIEW

**Comment by sgtcoolguy** [comment]
OK, after digging deeper - my changes definitely didn't cause the issue. Looks to be a main thread related issue: https://jira.appcelerator.org/browse/TIMOB-27076

Basically the bonjour APIs had locks on the main thread to wait for the service resolution (or publish/etc) to occur before returning (making the API sync).

As soon as I commented out the lock wait, it worked. The issue is that the API then becomes async and we'd need to fire an event to notify of success/failure to resolve/publish/etc. I don't know if we can explicitly run the resolution/publish off the main thread to "fix" this.

**Comment by janvennemann** [comment]
+1 on the Node style callbacks for the new async versions of mentioned functions. Since we are investing so much effort to increase node compatibility, our own API should also go this way (or promise as you already mentioned).

@sgtcoolguy judging by the PR description, i guess you last comment is obsolete now, and you already addressed that issue?

**Comment by sgtcoolguy** [comment]
For now I'm keeping both events *and* optional callbacks, so you can pick or choose which way to deal with things. I'd prefer callbacks myself, but most of our APIs are event based so it feels odd not to provide that for now.

**Review by vijaysingh-axway — CHANGES_REQUESTED** [review]
Few changes required as mentioned in comments.

File: iphone/Classes/TiNetworkBonjourServiceProxy.m [hunk 1]
```diff
@@ -294,27 +291,60 @@ - (void)setSocket:(TiNetworkTCPSocketProxy *)socket_
...
+  }
 }
 
 #pragma mark Resolution
 
 - (void)netService:(NSNetService *)service_ didNotResolve:(NSDictionary *)errorDict
 {
-  [self setError:[TiNetworkBonjourServiceProxy stringForErrorCode:[[errorDict valueForKey:NSNetServicesErrorCode] intValue]]];
+  // Fire event to notify of error resolving!
+  NSNetServicesError code = [errorDict valueForKey:NSNetServicesErrorCode];
```
[the remark this answers is not in the document]
Comment (sgtcoolguy) [reply iphone/Classes/TiNetworkBonjourServiceProxy.m]: `NSNetServicesError` is an enum of type `NSInteger`, so I don't think we need to do this.

**Comment by sgtcoolguy** [comment]
@vijaysingh-axway Pushed another commit to address your feedback (and another issue I saw internally with setSocket)

**Review by vijaysingh-axway — APPROVED** [review]
CR passed.

**Comment by lokeshchdhry** [comment]
FR Passed.

Checked above code with different `service type` & checked if `Titanium.Network.Socket.TCP` is affected.

Looks good.

Studio Ver:       6.0.0.201911251516
SDK Ver:          9.0.0 local build
OS Ver:           10.14.5
Xcode Ver:        Xcode 11.2.1
Appc NPM:         5.0.0-2
Appc CLI:         7.1.2
Daemon Ver:       1.1.3
Ti CLI Ver:       5.2.2
Alloy Ver:        1.14.4
Node Ver:         12.13.1
NPM Ver:          6.12.1
Java Ver:         11.0.1
IOS Simulator:    ⇨ iPhone 11 Pro Max (iOS-13-2)

**Pull request merged**